clients/src/main/java/org/apache/kafka/common/serialization/Deserializer.java

```diff
@@ -43,7 +45,20 @@
      * @param data serialized bytes; may be null; implementations are recommended to handle null by returning a value or null rather than throwing an exception.
      * @return deserialized typed data; may be null
      */
-    T deserialize(String topic, byte[] data);
+    default T deserialize(String topic, byte[] data) {
```

Comment:
hey @viktorsomogyi ,

Sorry I missed the KIP discussion, but I just read the KIP over and didn't see a explanation of this:

Can you explain the reason to provide a default implementation of this method?

Thanks,
-John

Reply:
Ah, I just now saw the discussion in the vote thread. It sounds like you're going to remove this default. Sorry about that.